packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -114,17 +117,29 @@ class BottomAppBar extends StatefulWidget {
   /// If this property is null then [BottomAppBarTheme.shape] of
   /// [ThemeData.bottomAppBarTheme] is used. If that's null then the shape will
   /// be rectangular with no notch.
+  ///
+  /// Ignored if the ambient [ThemeData.useMaterial3] is true; the Material 3
+  /// spec defines the shape without a notch:
+  ///   https://m3.material.io/components/bottom-app-bar/specs#9d91e4a4-a6d9-4ad5-afad-430267ebb4e9
   final NotchedShape? shape;
```

Comment:
This isn't just ignored in that case, but rather not permitted, right? So for example:

```suggestion
  /// If the ambient [ThemeData.useMaterial3] is true,
  /// then there is never a notch, and this property must be null.
  /// (The Material 3 spec does not provide for a notch:
  /// <https://m3.material.io/components/bottom-app-bar/specs#9d91e4a4-a6d9-4ad5-afad-430267ebb4e9>.)
  final NotchedShape? shape;
```